kickstarter/android-oss
# NT-1958: UX – Post a reply

# 📲 What

UX – Post a reply

# 🤔 Why

- to handle how we post a reply within a thread!

# 🛠 How
- Refactored ```PostCommentData```
- Refactored apolloclient apii to handle replies
```
... parentId(comment.parent?.let { encodeRelayId(it)})

```

# 👀 See


| Before 🐛 |
```
 this.service.mutate(
                CreateCommentMutation.builder()
                    .parentId(comment.parentId)
                    .commentableId(comment.commentableId)
                    .clientMutationId(comment.clientMutationId)
                    .body(comment.body)
                    .build()
            )
```
| After 🦋 |

```
 this.service.mutate(
                CreateCommentMutation.builder()
                    .parentId(comment.parent?.let { encodeRelayId(it)})
                    .commentableId(comment.commentableId)
                    .clientMutationId(comment.clientMutationId)
                    .body(comment.body)
                    .build()
            )
```
| --- | --- |
|  |  |

# 📋 QA
- Add this snippet to the  ``` ThreadViewmodel ``` init block
```

 /*
             * This block should be removed as its meant to test reply to a comment
             * The block above should show the replies
             */
            commentData
                .switchMap {
                    this.apolloClient.createComment(
                        PostCommentData(
                            commentableId = it.commentableId ?: "",
                            body = "Just a comment reply",
                            clientMutationId = null,
                            parent = it.comment
                        )
                    )
                }
                .compose(bindToLifecycle())
                .subscribe(
                    {
                        val c = it
                    },
                    {
                        val e = it
                    }
                )
```
- Go to any backed project comment screen
- Select any comment with replies. A reply is automatically posted 

# Story 📖

https://kickstarter.atlassian.net/browse/NT-1958
